src/crypto/Crypto.ts

```diff
@@ -224,51 +225,46 @@ export interface HashingOptions {
   hasher?: Hasher
 }
 
-const saltedBlake2b256: Hasher = (value, nonce) =>
+export const saltedBlake2b256: Hasher = (value, nonce) =>
   blake2AsHex((nonce || '') + value, 256)
 
 /**
- * Configurable computation of salted or unsalted hashes over an array or map of statements. Can be used to validate/reproduce salted hashes
+ * Configurable computation of salted over an array of statements. Can be used to validate/reproduce salted hashes
  * by means of an optional nonce map.
  *
- * @param statements A map or array of statement strings to be hashed. Keys will be used to look up nonces or as input for the `nonceGenerator`.
+ * @param statements An array of statement strings to be hashed.
  * @param options Optional hasher arguments.
- * @param options.nonces An optional map of nonces. If present, it must comprise all keys of `statements`, as those will be used map nonces to statements.
+ * @param options.nonces An optional map or array of nonces. If present, it should comprise all keys of `statements`, as those will be used map nonces to statements.
  * @param options.nonceGenerator An optional nonce generator. Will be used if `options.nonces` is not defined to generate a (new) nonce for each statement. The statement key is passed as its first argument. If no `nonces` or `nonceGenerator` are given this function returns unsalted hashes.
  * @param options.hasher The hasher to be used. Computes a hash from a statement and an optional nonce. Required but defaults to 256 bit blake2 over `${nonce}${statement}`.
- * @returns An array of objects for each statement which contain the hash, nonce and statement key.
+ * @returns An array of objects for each statement which contain a statement, its digest, salted hash and nonce.
  */
 export function hashStatements(
-  statements: string[] | Record<string, string>,
+  statements: string[],
   options: HashingOptions = {}
-): Array<{ key: string; hash: string; nonce?: string }> {
+): Array<{
+  digest: string
+  statement: string
+  saltedHash: string
+  nonce: string
+}> {
   // apply default hasher if undefined
-  const { hasher, nonces, nonceGenerator } = {
-    ...options,
-    hasher: options.hasher || saltedBlake2b256,
-  }
-  // getNonce <- if nonces map then take from map, else use nonceGenerator function
-  // if no nonceGenerator given, unsalted hashes are returned
+  const {
+    hasher = saltedBlake2b256,
```

Comment:
I prefer more explicit defaults

Reply:
I did a little research and this is actually the only concise vanilla js solution for the `key : undefined` problem.
We could thus either stick with it, use [`lodash.merge`](https://lodash.com/docs/4.17.15#merge) or go with something a little more ugly:
```js
  const defaults = {
    hasher : saltedBlake2b256,
    nonceGenerator : () => uuid(),
  }
  const {nonces} = options
  const hasher =  options.hasher || defaults.hasher
  const nonceGenerator = options.nonceGenerator || defaults.nonceGenerator
```

Reply:
(of course this would also work:
```js
  const {nonces} = options
  const hasher =  options.hasher || saltedBlake2b256
  const nonceGenerator = options.nonceGenerator || () => uuid()
```
but I wouldn't say this improves readability)

Reply:
I like the first proposal from you, where I added the like

Reply:
I think I have another favourite option: we purge the `options`:
```js
  // apply defaults
  Object.keys(options).forEach((key) => {
    if (typeof options[key] === 'undefined') delete options[key]
  })
  const defaults = {
    hasher: saltedBlake2b256,
    nonceGenerator: () => uuid(),
  }
  const { nonces, nonceGenerator, hasher } = { ...defaults, ...options }
```

Reply:
I think it is a bit overkill, just for applying defaults.

Reply:
I like the proposal too with the likes, though I don't see how it's clearer than:

```js
const {hasher : saltedBlake2b256,
    nonceGenerator : () => uuid(),
    } = { 
    hasher: undefined,
    nonceGenerator: undefined
    }

return {hasher, nonceGenerator}
```